### PR TITLE
feature(notify): support RingCentral webhook

### PR DESCRIPTION
**Pull request by onyas**
### Add new notify: RingCentral Webhook. 

Please check here for how to create a webhook in RingCentral. https://developers.ringcentral.com/guide/team-messaging/incoming-webhooks/webhook-creation.

### Some screenshots for the message format

- Failure notify

image

- Recovery notify

image

- SLA report

image

Please help review, thanks.

### CODE REVIEW

**Comment by samanhappy** [comment]
Thanks very much for this PR, We would really appreciate it if you update the docs(README.md, Manual.md, etc.) accordingly.

**Comment by onyas** [comment]
Yeah, I just updated the docs. please let me know if I miss anything. 

Thanks.

File: notify/ringcentral/ringcentral.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+func (c *NotifyConfig) Config(gConf global.NotifySettings) error {
+	c.NotifyKind = "ringcentral"
+	c.NotifyFormat = report.Text
+	c.NotifySendFunc = c.SendRingCentral
+	c.DefaultNotify.Config(gConf)
+	log.Debugf("Notification [%s] - [%s] configuration: %+v", c.NotifyKind, c.NotifyName, c)
+	return nil
+}
+
+func (c *NotifyConfig) SendRingCentral(title, msg string) error {
```
Comment (haoel) [inline notify/ringcentral/ringcentral.go:50]: Please fix this lint warning.

**Review by haoel — CHANGES_REQUESTED** [review]
Thanks for this PR, The code is clean and have unit tests, and updated documents and resource files. That's Cool!

just one minor thing - please fix the code lint warning.

you can run `revive ./...` to see the code lint warning in your local

File: notify/ringcentral/ringcentral.go (hunk 1 above)
Comment (onyas) [reply notify/ringcentral/ringcentral.go:52]: Add comments for the exported method and type. Please help review.

Thanks.

**Review by haoel — APPROVED** [review]

**Comment by haoel** [comment]
Thanks for this PR, well done! would you please leave your personal contact info to me( send to haoel@hotmail.com ) ? @onyas 

BTW, the JSON schema would be updated by Github Action after the PR merged. but let's see what would be happened if you updated it in PR.

**Pull request merged**